src/types/hexadecimal.js

```diff
@@ -0,0 +1,170 @@
+import * as validate from './validate'
+
+/**
+ * Encoder
+ *
+ * @param {string} str string to encode
+ * @param {Array} buffer buffer to place result to
+ * @param {number} from position to write from
+ * @returns {Array} modified buffer
+ * @private
+ */
+function insertHexadecimalToByteArray (str, buffer, from) {
+  for (let i = 0; i < str.length; i += 2) {
+    buffer[from] = parseInt(str.substr(i, 2), 16)
+    from++
+  }
+  return buffer
+}
+
+/**
+ * Validator wrapper
+ *
+ * @param {string} name structure name
+ * @param {number} size value size in bytes
+ * @param {string} value value representation
+ * @returns {string} value if validation passes
+ * @throws {TypeError} in case of validation break
+ * @private
+ */
+function validateHexadecimal (name, size, value) {
+  if (!validate.validateHexadecimal(value, size)) {
+    throw new TypeError(`${name} of wrong type is passed: ${value}`)
+  }
+
+  return value
+}
+
+/**
+ * Factory for building Hex Types
+ *
+ * @param {function(value, buffer, from)} serizalizer function accepting value, buffer, position and returns modified buffer
+ * @param {number} size type size in bytes
+ * @param {string} name type name to distinguish between types
+ * @returns {Object} hex type
+ */
+function hexTypeFactory (serizalizer, size, name) {
+  return Object.defineProperties({}, {
+    size: {
```

Comment:
Could seemingly be rewritten with

```javascript
function hexTypeFactory (serizalizer, size, name) {
  return {
    get size() { return size; },
    // ...
  }
}
```

Also, pretty controversial, but I think that using `Object.defineProperties` in this situations more leads to complicated code than brings type safety.

Reply:
It is not exactly the same, getters can be deleted. It's the only place where definitions are used.